README.md

```diff
@@ -65,6 +67,15 @@ bazel-bin/nighthawk_client  [--trace <uri format>]
 
 Where:
 
+--failure-predicate <<string, uint32_t>>  (accepted multiple times)
+Failure predicate. Allows specifying a counter name plus threshold
```

Comment:
Do thresholds cover all cases? I'm curious if we might want to consider using something more powerful, like CEL (already in the Envoy dependency tree due to its use in RBAC), or some other custom expression format.

Reply:
I think you are right in that it is not hard to come up with exit conditions that would be useful, but can't be expressed with just the chaining of thresholds for absolute counter values we accomplish here. So I was hoping this current PR state embodies a reasonable start, by capturing some necessities with a simple approach. Next to being able to express more complex logical conditions, arguably more input to the evaluating those could be useful: latency, delta's of counters / gauges, maybe even environment variables for simple signalling from tested application code.

Glossing over the CEL / RBAC implementation you point out, I think that looks like a great opportunity to enhance this. If we'd want move towards that, we can probably do so without breaking existing setups, by implementing (a) new type(s) of `TerminationPredicate`?

Reply:
Let's ship with what you have. The user community is small enough that I think we can cope with evolution of this.